crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -80,13 +87,32 @@ impl Auction {
                             fee: bigdecimal::BigDecimal::new(3.into(), 3),
                         })
                     }
-                    liquidity::Kind::UniswapV3(_) => todo!(),
+                    liquidity::Kind::UniswapV3(pool) => {
+                        tokens.insert(pool.tokens.get().0.into(), Default::default());
+                        tokens.insert(pool.tokens.get().1.into(), Default::default());
+                        Liquidity::ConcentratedLiquidity(ConcentratedLiquidityPool {
+                            id: liquidity.id.into(),
+                            address: pool.address.0,
+                            gas_estimate: liquidity.gas.0,
+                            tokens: vec![pool.tokens.get().0.into(), pool.tokens.get().1.into()],
+                            sqrt_price: pool.sqrt_price.0,
+                            liquidity: pool.liquidity.0,
+                            tick: pool.tick.0,
+                            liquidity_net: pool
+                                .liquidity_net
+                                .iter()
+                                .map(|(key, value)| (key.0, value.0))
+                                .collect(),
+                            fee: rational_to_big_decimal(&pool.fee.0),
+                        })
+                    }
                     liquidity::Kind::BalancerV2Stable(_) => todo!(),
                     liquidity::Kind::BalancerV2Weighted(_) => todo!(),
                     liquidity::Kind::Swapr(_) => todo!(),
                     liquidity::Kind::ZeroEx(_) => todo!(),
                 })
                 .collect(),
+            tokens,
```

Comment:
Nit: I don't think we should rely on the order of field initialization. I think the code is correct here, but I vaguely remember this being a `clippy` lint...

I may be wrong though (maybe @vkgnosis knows).

Reply:
The order of initialization is defined to behave as the code expects it here. Changing this would be a breaking change requiring a new edition also the borrow checker would complain here.